nautobot/extras/models/tags.py

```diff
@@ -88,3 +98,55 @@ class TaggedItem(BaseModel, GenericUUIDTaggedItemBase):
 
     class Meta:
         index_together = ("content_type", "object_id")
+
+
+# 2.0 TODO: remove this, force migration to the newer django-taggit API.
+class _NautobotTaggableManager(_TaggableManager):
+    """Extend _TaggableManager to work around a breaking API change between django-taggit 1.x and 2.x.
+
+    This is a bit confusing, as there's also a related `TaggableManager` class as well.
+    `TaggableManager` is the *model field* (subclass of `models.fields.related.RelatedField`),
+    while `_TaggableManager` is the *associated manager* (subclass of `models.Manager`).
+
+    For `TaggableManager`, we chose to monkey-patch rather than subclass to override its `formfield` method;
+    replacing it with a subclass would create database migrations for every PrimaryModel with a `tags` field.
+    In 2.0 we'll want to bite the bullet and make the cutover (#1633).
+
+    For `_TaggableManager`, we can subclass rather than monkey-patching because replacing it *doesn't* require
+    a database migration, and this is cleaner than a monkey-patch.
+    """
+
+    def set(self, *tags, through_defaults=None, **kwargs):
+        """
+        Patch model.tags.set() to be backwards-compatible with django-taggit 1.x and forward-compatible with later.
+
+        Both of these approaches are supported:
+
+        - tags.set("tag 1", "tag 2")  # django-taggit 1.x
+        - tags.set(["tag 1", "tag 2"])  # django-taggit 2.x and later
+        """
+        if len(tags) == 1 and not isinstance(tags[0], (self.through.tag_model(), str)):
+            # taggit 2.x+ style, i.e. `set([tag, tag, tag])`
+            tags = tags[0]
+        else:
+            # taggit 1.x style, i.e. `set(tag, tag, tag)`
+            # Note: logger.warning() only supports a `stacklevel` parameter in Python 3.8 and later
+            tags_unpacked = ", ".join([repr(tag) for tag in tags])
+            tags_list = list(tags)
+            message = "Deprecated `tags.set(%s)` was called, please change to `tags.set(%s)` instead"
+            if sys.version_info >= (3, 8):
+                logger.warning(message, tags_unpacked, tags_list, stacklevel=2)
+            else:  # Python 3.7
+                logger.warning(message, tags_unpacked, tags_list)
+        return super().set(tags, through_defaults=through_defaults, **kwargs)
+
+
+class TaggedModel(PrimaryModel):
+    """
+    Abstract model which adds `tags` to the model
+    """
+
+    tags = TaggableManager(through=TaggedItem, manager=_NautobotTaggableManager, ordering=["name"])
+
+    class Meta:
+        abstract = True
```

Comment:
I don't quite understand this change. The only difference between `PrimaryModel` and `OrganizationalModel` was that the former had tags - now `PrimaryModel` and `OrganizationalModel` are identical classes, and we have to use `TaggedModel` all over the place in place of `PrimaryModel`?

Reply:
Yr `PrimaryModel` has a dependency on `extras.tags`, which was causing circular import errors everywhere. The only solution was to move `PrimaryModel` to `extras.tags`, but I decided to create the `TaggedModel` instead. 
I didn't remove the `PrimaryModel` because I taught it had a different purpose than `OrganizationModel` based on the docstring, but if `PrimaryModel` is useless, I could remove it.

Reply:
IMO there is no point to differentiating them any longer. Every model should support `tags`.